app/org/sagebionetworks/bridge/cache/CacheProvider.java

```diff
@@ -26,41 +34,72 @@ public void setJedisOps(JedisOps jedisOps) {
         this.jedisOps = jedisOps;
     }
 
-    public void setUserSession(String key, UserSession session) {
+    public void setUserSession(final UserSession session) {
+        checkNotNull(session);
+        List<Object> results = null;
         try {
-            String ser = BridgeObjectMapper.get().writeValueAsString(session);
-            String redisKey = RedisKey.SESSION.getRedisKey(key);
-            String result = jedisOps.setex(redisKey, BridgeConstants.BRIDGE_SESSION_EXPIRE_IN_SECONDS, ser);
+            final String userId = session.getUser().getId();
+            final String userKey = RedisKey.USER_SESSION.getRedisKey(userId);
+            final String sessionToken = session.getSessionToken();
+            final String sessionKey = RedisKey.SESSION.getRedisKey(sessionToken);
+            final String ser = BridgeObjectMapper.get().writeValueAsString(session);
+            results = jedisOps.getTransaction()
+                    .setex(userKey, BridgeConstants.BRIDGE_SESSION_EXPIRE_IN_SECONDS, sessionToken)
```

Comment:
It looks like we already do this:
https://github.com/Sage-Bionetworks/BridgePF/blob/develop/app/controllers/AuthenticationController.java#L29

Eric, can you explicitly test this? Ideally in an integration test, but a manual test will suffice for now.